### ISSUE TITLE
Results depend on dictionary order

### ISSUE THREAD

**Issue by klipspringr** [goingonit/wordle#1]
Thanks for uploading this, it's been really interesting getting my head round it!

I noticed that results depend on dictionary order. Running wordle.ts on the "official" Wordle solutions list from the game js, I get:
- original order: [ 'arise', 'hotel', 'gawky', 'rover', 'cover', 'mover' ] (6)
- sorted a-z: [ 'arise', 'hotel', 'gawky', **'corer', 'boxer', 'mover' ]** (6)
- sorted z-a: **[ 'raise', 'trend', 'wreck', 'freak', 'break' ]** (5)

So the length of the longest chain depends on dictionary ordering. On the first guess, ARISE and RAISE score equally. My understanding of the Knuth heuristic is that, in this case, the lowest-numeric (valid) guess is chosen. So RAISE would never be played.

Word list taken from the game source is [here for ease of ref](https://github.com/klipspringr/wordle-cli/tree/main/data).

**Comment by goingonit** [comment]
Yes indeed, there should be a canonical ordering! I also am not so sure that there aren't larger bugs lurking here, more testing is definitely necessary. Finally, you can imagine better heuristics for choosing a guess - I am skeptical that `raise -> trend` and `arise -> hotel` are truly indistinguishable in quality, maybe one gives you a higher probability of green squares? Unfortunately I just started a new job today so I might not have time to take a look until the weekend.

Glad you are finding the code interesting, it was fun to write. (though optimizing Node code is not my favorite -- "all numbers are doubles but you can do bit arithmetic but only for 32 bits" is a silly semantic and it makes me sad that I am bit twiddling at all but there's like a 5x runtime difference compared to storing an array of counts!)
